app.py

```diff
@@ -277,179 +256,178 @@ def parse_repository_file_response(file_response: dict[str, Any]) -> str:
+                "created_on": convert_to_datetime(pr.get("createdDate")),
+                "updated_on": convert_to_datetime(pr.get("updatedDate")),
+                "merge_commit": pr.get("fromRef", {}).get("latestCommit"),
                 "description": pr.get("description"),
-                "state": pr["state"],
-                "owner": pr["author"]["user"]["displayName"],
-                "link": pr["links"]["self"][0]["href"],
-                "destination": pr["toRef"]["displayId"],
+                "state": pr.get("state"),
+                "owner": pr.get("author", {}).get("user", {}).get("displayName"),
+                "link": pr.get("links", {}).get("self", [{}])[0].get("href"),
+                "destination": pr.get("toRef", {}).get("displayId"),
                 "reviewers": [
-                    user["user"]["displayName"] for user in pr.get("reviewers", [])
+                    user.get("user", {}).get("displayName") for user in pr.get("reviewers", [])
                 ],
-                "source": pr["fromRef"]["displayId"],
+                "source": pr.get("fromRef", {}).get("displayId"),
             },
             "relations": {
                 "repository": pr["toRef"]["repository"]["slug"],
-                "participants": [pr.get("author")["user"]["emailAddress"]]
-                + [user["user"]["emailAddress"] for user in pr.get("participants", [])],
+                "participants": [pr.get("author", {}).get("user", {}).get("emailAddress")]
+                                + [user.get("user", {}).get("emailAddress") for user in pr.get("participants", [])],
             },
         }
-        add_entity_to_port(blueprint_id=blueprint_id, entity_object=entity)
+        await add_entity_to_port(blueprint_id=blueprint_id, entity_object=entity)
 
-
-def get_repository_readme(project_key: str, repo_slug: str) -> str:
+async def get_repository_readme(project_key: str, repo_slug: str) -> str:
     file_path = f"projects/{project_key}/repos/{repo_slug}/browse/README.md"
     readme_content = ""
-    for readme_file_batch in get_paginated_resource(
-        path=file_path, page_size=500, full_response=True
+    async for readme_file_batch in get_paginated_resource(
+            path=file_path, page_size=500, full_response=True
     ):
         file_content = parse_repository_file_response(readme_file_batch)
         readme_content += file_content
     return readme_content
 
-
-def get_latest_commit(project_key: str, repo_slug: str) -> dict[str, Any]:
+async def get_latest_commit(project_key: str, repo_slug: str) -> dict[str, Any]:
     try:
         commit_path = f"projects/{project_key}/repos/{repo_slug}/commits"
-        for commit_batch in get_paginated_resource(path=commit_path, page_size=1):
+        async for commit_batch in get_paginated_resource(path=commit_path, page_size=1):
             if commit_batch:
                 latest_commit = commit_batch[0]
                 return latest_commit
```

Comment:
I just realized that this may cause the app to be slow. Assuming a repository has 5k commits, once we query this api, we pass a page size of 1. this means that it will make 5k API calls. and that could be why the app is slow. We need to find a way to get this data efficiently. maybe instead of paginated, we just make a single API call since the result is already sorted by most recent?

<img width="900" alt="Screenshot 2024-09-13 at 9 33 51 AM" src="https://github.com/user-attachments/assets/9f601996-1ad9-4909-b3bc-80eb0e10b7ec">